### PR TITLE
Add metric deleting_pods_total

### PR DESCRIPTION
**Pull request by xing-yang**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR adds metrics `force_delete_pods_total` and `force_delete_pod_errors_total` in the Pod GC Controller.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
Adds metrics `force_delete_pods_total` and `force_delete_pod_errors_total` in the Pod GC Controller.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/podgc/metrics.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			Help:           "Number of errors encountered when forcefully deleting the pods since the Pod GC Controller started.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register the metrics that are to be monitored.
```
Comment (gnufied) [inline pkg/controller/podgc/metrics.go:55]: Usually metrics are registered in `func init(){}`call.

File: pkg/controller/podgc/metrics.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	podGCController   = "pod_gc_collector"
+	deletingPods      = "deleting_pods_total"
```
Comment (gnufied) [inline pkg/controller/podgc/metrics.go:28]: should we call it "force_deleted_pods_total" and "force_delete_pod_errors_total" ?

I would prefer to have word `force` somewhere in the name.

**Review by gnufied — COMMENTED** [review]
Also cc @kubernetes/sig-instrumentation-members

File: pkg/controller/podgc/metrics.go (hunk 2 above)
Comment (logicalhan) [reply pkg/controller/podgc/metrics.go:28]: I like those suggestions.

File: pkg/controller/podgc/metrics.go (hunk 1 above)
Comment (logicalhan) [reply pkg/controller/podgc/metrics.go:55]: This is also a common pattern.

**Review by logicalhan — COMMENTED** [review]
/lgtm

(from sig inst)

File: pkg/controller/podgc/metrics.go (hunk 1 above)
Comment (xing-yang) [reply pkg/controller/podgc/metrics.go:55]: Sure.

File: pkg/controller/podgc/metrics.go (hunk 2 above)
Comment (xing-yang) [reply pkg/controller/podgc/metrics.go:28]: That makes sense.

**Comment by xing-yang** [comment]
@gnufied @logicalhan Comments are addressed.  PTAL.  Thanks.

File: pkg/controller/podgc/metrics.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	podGCController   = "pod_gc_collector"
+	deletingPods      = "force_delete_pods_total"
```
Comment (logicalhan) [inline pkg/controller/podgc/metrics.go:28]: I would just inline these names in the metric definition, it's confusing to jump around to figure out what the name of the metric actually is.

File: pkg/controller/podgc/metrics.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register the metrics that are to be monitored.
+func Register() {
```
Comment (gnufied) [inline pkg/controller/podgc/metrics.go:56]: nit: Can we call this function `RegisterMetrics` ?

File: pkg/controller/podgc/metrics.go (hunk 3 above)
Comment (xing-yang) [reply pkg/controller/podgc/metrics.go]: Done

File: pkg/controller/podgc/metrics.go (hunk 4 above)
Comment (xing-yang) [reply pkg/controller/podgc/metrics.go]: Done

**Comment by xing-yang** [comment]
@gnufied @logicalhan Addressed your comments. PTAL. Thanks.

**Review by logicalhan — COMMENTED** [review]
/lgtm
(from sig instrumentation)

**Comment by dims** [comment]
at least 3 lgtm(s). shipping it! changes look good to me as well.

/approve
/lgtm

**Pull request merged**